storagenode/nodestats/service.go

```diff
@@ -66,21 +43,160 @@ func (c *Client) Close() error {
 	return &Service{
-		log:       log,
-		transport: transport,
-		kademlia:  kademlia,
+		log:         log,
+		statsTicker: time.NewTicker(time.Duration(time.Second * 30)),
+		spaceTicker: time.NewTicker(time.Duration(time.Second * 60)),
+		transport:   transport,
+		consoleDB:   consoleDB,
+		kademlia:    kademlia,
+	}
+}
+
+// RunStatsLoop continuously queries satellite for stats info with some interval
+func (s *Service) RunStatsLoop(ctx context.Context) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	for {
+		err = s.CacheStatsFromSatellites(ctx)
+		if err != nil {
+			s.log.Error(fmt.Sprintf("Get stats query failed: %v", err))
+		}
+
+		select {
+		// handle cancellation signal first
+		case <-ctx.Done():
+			return ctx.Err()
+		// wait for the next interval to happen
+		case <-s.statsTicker.C:
+		}
+	}
+}
+
+// CacheStatsFromSatellites queries node stats from all the satellites
+// known to the storagenode and stores this information into db
+func (s *Service) CacheStatsFromSatellites(ctx context.Context) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	satellites, err := s.consoleDB.GetSatelliteIDs(ctx, time.Time{}, time.Now())
+	if err != nil {
+		return NodeStatsServiceErr.Wrap(err)
+	}
+
+	var cacheStatsErr errs.Group
+	for _, satellite := range satellites {
+		stats, err := s.GetStatsFromSatellite(ctx, satellite)
+		if err != nil {
+			cacheStatsErr.Add(err)
+			continue
+		}
+
+		stats.UpdatedAt = time.Now()
+
+		// try to update stats from satellite
+		if err = s.consoleDB.UpdateStats(ctx, *stats); err != nil {
+			// if stats doesn't exists - create new one
+			if err == sql.ErrNoRows {
+				_, err = s.consoleDB.CreateStats(ctx, *stats)
+				if err != nil {
+					cacheStatsErr.Add(NodeStatsServiceErr.Wrap(err))
+					continue
+				}
+			}
+
+			cacheStatsErr.Add(NodeStatsServiceErr.Wrap(err))
+			continue
+		}
+
+		s.log.Info(fmt.Sprintf("CacheStats %s: %v", satellite, stats))
+
+		satStats, err := s.consoleDB.GetStatsSatellite(ctx, satellite)
+		if err != nil {
+			s.log.Error(fmt.Sprintf("SAT %s err: %v", satellite, err))
+		}
+		s.log.Info(fmt.Sprintf("CacheStats QUERY SAT %s: %v", satellite, satStats))
+	}
+
+	return cacheStatsErr.Err()
+}
+
+// RunStatsLoop continuously queries satellite for disk space usage with some interval
+func (s *Service) RunSpaceLoop(ctx context.Context) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	for {
+		err = s.CacheSpaceUsageFromSatellites(ctx)
+		if err != nil {
+			s.log.Error(fmt.Sprintf("Get disk space usage query failed: %v", err))
+		}
+
+		select {
+		// handle cancellation signal first
+		case <-ctx.Done():
+			return ctx.Err()
+		// wait for the next interval to happen
+		case <-s.statsTicker.C:
+		}
 	}
 }
 
+// CacheSpaceUsageFromSatellites queries disk space usage from all the satellites
+// known to the storagenode and stores this information into db
+func (s *Service) CacheSpaceUsageFromSatellites(ctx context.Context) (err error) {
```

Comment:
Why isn't there only a single ticker loop?

Reply:
We discussed that we should have to separate intervals, as nodestats is updated more frequently and is less costly than rollups query

Reply:
Yeah, i agree having two is better in this workflow!